### DISCUSSION TITLE
cadvisor doesn't provide container stats...

### DISCUSSION THREAD

**Discussion by anubhavjaiswal03** [mbround18/valheim-docker#574] [Help!] [answered]
followed the instructions to the letter,  I found major. issues with cadvisor,

1: the port 8080 was not exposed on the container
2: checking the log this is the message:

F0509 06:07:58.663998       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:07:59.286997       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:07:59.883954       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:08:00.709888       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:08:01.991098       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:08:04.115772       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:08:07.748380       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:08:14.645485       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:08:27.900353       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:08:54.024360       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:09:45.687670       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:10:46.152648       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:11:46.687632       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:12:47.132101       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:13:47.649019       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:14:48.111860       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found
F0509 06:15:48.587327       1 cadvisor.go:146] Failed to create a Container Manager: mountpoint for cpu not found

please help me :)

**Comment by anubhavjaiswal03** [comment, marked as answer]
I resolved the above issues it seems to be a issue with      - /dev/disk:/dev/disk:ro
removing it boot up cadvisor. the port were exposed properly.
but now I am not able to see docker stats like CPU Memory in Grafana for your Dashboard: 14123

**Reply by anubhavjaiswal03** [reply]
Still unable to see the CPU Usage Metrics
<img width="1792" alt="Screenshot 2022-05-09 at 1 12 10 PM" src="https://user-images.githubusercontent.com/36488000/167363487-e326402b-1f22-44ab-944b-800a743136a2.png">

**Reply by mbround18** [reply]
@anubhavjaiswal03 thats really weird, what kinda system are you running this on?

**Reply by mbround18** [reply]
I should have a direct metric you could build a graphana dashboard off of, https://github.com/mbround18/valheim-docker/discussions/330
